sample-node-config/aws/README.md

```diff
@@ -0,0 +1,234 @@
+
+This directory contains some [Terraform](https://www.hashicorp.com/products/terraform) and AWS based infrastructure code to setup a single [Hydra node](https://hydra.family/head-protocol/docs/getting-started/installation) connected to a [Cardano node](https://docs.cardano.org/getting-started/installing-the-cardano-node) running on `preview` testnet. It's not a complete turnkey solution and requires some tweaking and parameterisation to be completely usable but we thought it would be good starting point for new Hydra users.
+
+## Configuring Hydra Node
+The [configuration script](./scripts/configure-testnet.sh) assumes this image must be a reasonably recent Ubuntu/Debian Linux distribution, with [docker](https://docker.io) and [docker-compose](https://docs.docker.com/compose/) installed, and a configured user `ubuntu` with `sudo` access.
+
+The configuration process expects to find some files which are not provided by default and which are required for starting the Hydra node:
+* A Hydra signing key file `hydra-key.sk` which will be used in the Head to sign snapshots.
+  This can be generated using [hydra-tools](https://hydra.family/head-protocol/docs/getting-started/quickstart#hydra-keys),
+* A cardano signing key file  `cardano-key.sk` which is required to identify the parties on-chain and sign transactions.
+  This is a standard Cardano key so one can reuse an existing key or [generate a new one](https://hydra.family/head-protocol/docs/getting-started/quickstart#cardano-keys),
+* 0 or more hydra verification keys and cardano verification keys for the other Head parties,
+* The IP addresses and ports of _peer_ nodes,
+* Configuration files for [promtail](https://grafana.com/docs/loki/latest/clients/promtail/) and [prometheus](https://prometheus.io/) which are run as part of the stack,
+* Configuration files for the off-chain ledger.
+
+The key files should be put in the current directory and their name referenced in the [testnet.tf](./testnet.tf) file. Then the [docker-compose.yaml](./docker/docker-compose.yaml) should be edited to reflect the above parameters as [command-line arguments](https://hydra.family/head-protocol/docs/getting-started/quickstart) to the `hydra-node` container.
+
+The [promtail-config.yml](./docker/promtail-config.yml) should be edited to point to the correct URL where logs should be shipped or the promtail container altogether removed.
+
+### Pre-requisites
+- you have access to an aws account with root priviledges.
+- you have configured your local aws credentials. 
+for this example in `~/.aws/credentials` we have:
+    ```
+    [personal]
+    aws_access_key_id=???
+    aws_secret_access_key=???
+    region=eu-west-3
+    ```
+> **personal** will be your AWS_PROFILE
+> so whenever you see _personal_ in the rest of the file you need to replace it with your own aws profile name.
+> Also you need to make sure to create a new key pair in the region eu-west-3 (TODO: abstract the region away?)
```

Comment:
Do not change anything here now but, maybe, in the future, it would be cool to use aws_ssm to connect to the EC2 so that we don't depend on extra ssh keys manipulations.

Reply:
adding TODO